cloud/amazon/ec2_ami.py

```diff
@@ -393,7 +393,8 @@ def create_image(module, ec2):
         try:
             img = ec2.get_image(image_id)
 
-            if img.state == 'available':
+            # We must wait for the ami to exist in AWS api responses before we can skip the wait, and add tags when specifying wait: no 
+            if img.state == 'available' or (img and not wait):
```

Comment:
The reason why we appear to getting the "Error while trying to find the new image..." error is that while the change to this line allows for breaking out of the for loop when `img.state` is not `available`, there is a check right immediately after this `for` loop on line 405 where the module fails with this error message `if img.state != 'available'`.